improver/utilities/mask.py

```diff
@@ -0,0 +1,71 @@
+# (C) Crown copyright, Met Office. All rights reserved.
+#
+# This file is part of IMPROVER and is released under a BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module for applying mask to a cube."""
+
+from typing import Union
+
+import iris
+import numpy as np
+
+from improver.utilities.common_input_handle import as_cubelist
+from improver.utilities.cube_checker import find_dimension_coordinate_mismatch
+from improver.utilities.cube_manipulation import (
+    enforce_coordinate_ordering,
+    get_coord_names,
+)
+
+
+def apply_mask(
+    *cubes: Union[iris.cube.CubeList, iris.cube.Cube],
+    mask_name: str,
+    invert_mask: bool = False,
+) -> iris.cube.Cube:
+    """
+    Apply a provided mask to a cube. If invert_mask is True, the mask will be inverted.
+
+    Args:
+        cubes:
+            A list of iris cubes that should contain exactly two cubes: a mask and a cube
+            to apply the mask to. The cubes should have the same dimensions.
+        mask_name:
+            The name of the mask cube. It should match with exactly one of the cubes in
+            the input cubelist.
+        invert_mask:
+            If True, the mask will be inverted before it is applied.
+    Returns:
+        A cube with a mask applied to the data.
+
+    Raises:
+    ValueError: If the number of cubes provided is not equal to 2.
+
+    """
+    cubes = as_cubelist(*cubes)
+    cube_names = [cube.name() for cube in cubes]
+    if len(cubes) != 2:
+        raise ValueError(
+            f"""Two cubes are required for masking, a mask and the cube to be masked.
+                         Provided cubes are {cube_names}"""
+        )
+
+    mask = cubes.extract_cube(mask_name)
+    cubes.remove(mask)
+    cube = cubes[0]
```

Comment:
I was thinking that the inputs would be a mask and the cube you want to mask. In theory you could make the plugin work with multiple cube (i.e. you input a mask and a list of cubes you want to apply the mask to and apply the mask to each input cube) but then you'd have to output all the cubes that you have. As we (mostly) require plugins to only output one cube it would be odd to make an exceptionfor this plugin.

Reply:
> As we (mostly) require plugins to only output one cube it would be odd to make an exceptionfor this plugin.

Don't worry about it now.  I think this is something worth considering in future.  I took such an approach with some plugins as part of the CLI/plugin refactor work.  Considering a plugin as being an operator, taking 1 or more cube to apply to that operator doesn't preclude it from being used with only 1 and extending its support would offer more options for graph construction.